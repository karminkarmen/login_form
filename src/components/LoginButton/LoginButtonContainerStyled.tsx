import styled from 'styled-components';
import { centered } from '../../configs/mixins';
import { innerWidthSize } from '../../constants/themeConsts';

const LoginButtonContainerStyled = styled.div`
    ${centered}
    position: relative;
    width: ${innerWidthSize / 2}px;
    height: ${innerWidthSize / 2}px;
    left: ${innerWidthSize + 10}px;
    top: -130px;
    border-radius: ${props => props.theme.radius.maximum};
    background: ${props => props.theme.colors.main};
    transition: ${props => props.theme.transitionTime.default};

    &:hover {
        background-color: ${props => props.theme.colors.secondaryLighten};
        cursor: pointer;
    }

    &:focus {
        outline: none;
    }
`;

export default LoginButtonContainerStyled;
