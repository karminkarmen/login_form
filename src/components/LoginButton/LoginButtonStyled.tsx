import styled from 'styled-components';
import LoginButtonContainerStyled from './LoginButtonContainerStyled';
import { innerWidthSize } from '../../constants/themeConsts';

const LoginButtonStyled = styled.input.attrs(props => ({
    type: 'submit',
    value: 'login'
}))`
    width: ${innerWidthSize / 2}px;
    height: ${innerWidthSize / 2}px;
    border-radius: ${props => props.theme.radius.maximum};
    border: unset;
    text-transform: uppercase;
    letter-spacing: ${props => props.theme.letterSpacing.normal};
    background: transparent;
    color: ${props => props.theme.colors.secondary};
    font-size: ${props => props.theme.fontSizes.medium};
    font-family: ${props => props.theme.fontFamilies.lato};
    transition: ${props => props.theme.transitionTime.default};
    background: ${props => props.theme.colors.main};

    ${LoginButtonContainerStyled}:hover & {
        color: ${props => props.theme.colors.main};
        background-color: ${props => props.theme.colors.secondaryLighten};
        cursor: pointer;
    }
    &:focus {
        outline: none;
    }
`;

export default LoginButtonStyled;
