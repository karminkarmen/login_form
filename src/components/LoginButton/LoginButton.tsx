import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import LoginButtonStyled from './LoginButtonStyled';
import LoginButtonContainerStyled from './LoginButtonContainerStyled';

type Props = {
    btnHandler: () => void;
}

const LoginButton: React.FC<Props> = ({btnHandler}) => {
    return (
        <ThemeProvider theme={theme}>
            <LoginButtonContainerStyled onClick={btnHandler}>
                <LoginButtonStyled />
            </LoginButtonContainerStyled>
        </ThemeProvider>
    );
};

export default LoginButton;