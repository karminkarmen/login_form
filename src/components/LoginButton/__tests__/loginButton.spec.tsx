import React from 'react';
import { shallow } from 'enzyme';

import LoginButton from '../LoginButton';
import LoginButtonContainerStyled from '../LoginButtonContainerStyled';
import LoginButtonStyled from '../LoginButtonStyled';

describe('<LoginButton />', () => {

    it('matches snapshot', () => {
        const wrapper = shallow(<LoginButton />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should render login button container', () => {
        const wrapper = shallow(<LoginButton />);

        expect(wrapper.find(LoginButtonContainerStyled)).toHaveLength(1);
    });

    it('should render styled login button', () => {
        const wrapper = shallow(<LoginButton />);

        expect(wrapper.find(LoginButtonStyled)).toHaveLength(1);
    });

});
