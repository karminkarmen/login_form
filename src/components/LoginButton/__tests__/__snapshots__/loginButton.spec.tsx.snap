// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> matches snapshot 1`] = `
"<ThemeProvider theme={{...}}>
  <styled.div>
    <styled.input />
  </styled.div>
</ThemeProvider>"
`;
